bench/macro/rocksdb-bench-wp8.hs

```diff
@@ -0,0 +1,418 @@
+        opts.seed
+
+-------------------------------------------------------------------------------
+-- Batch generation
+-------------------------------------------------------------------------------
+
+{- | Implement generation of unbounded sequence of insert/delete operations
+
+matching UTxO style from spec: interleaved batches insert and lookup
+configurable batch sizes
+1 insert, 1 delete, 1 lookup per key.
+
+Current approach is probabilistic, but uses very little state.
+We could also make it exact, but then we'll need to carry some state around
+(at least the difference).
+
+-}
+generateBatch
+    :: Int       -- ^ initial size of the collection
+    -> Int       -- ^ batch size
+    -> MCG.MCG   -- ^ generator
+    -> Int       -- ^ batch number
+    -> (MCG.MCG, [Word64], [Word64])
+generateBatch initialSize batchSize g b = (nextG, lookups, inserts)
+  where
+    maxK :: Word64
+    maxK = fromIntegral $ initialSize + batchSize * b
+
+    lookups :: [Word64]
+    (!nextG, lookups) = mapAccumL (\g' _ -> swap (MCG.reject maxK g')) g [1 .. batchSize]
+
+    inserts :: [Word64]
+    inserts = [ maxK .. maxK + fromIntegral batchSize - 1 ]
+
+-------------------------------------------------------------------------------
+-- run
+-------------------------------------------------------------------------------
+
+doRun :: GlobalOpts -> RunOpts -> IO ()
+doRun gopts opts = do
+    removePathForcibly $ gopts.rootDir ++ "_cp"
+    makeCheckpoint gopts
+
+    time <- timed_ $ doRun' gopts { rootDir = gopts.rootDir ++ "_cp" } opts
+    -- TODO: collect more statistic, save them in dry-run,
+    -- TODO: make the results human comprehensible.
+    printf "Proper run:            %7.03f sec\n" time
+    let ops = opts.batchCount * opts.batchSize
+    printf "Operations per second: %7.01f ops/sec\n" (fromIntegral ops / time)
+
+makeCheckpoint :: GlobalOpts -> IO ()
+makeCheckpoint gopts =
+    rocksDbWithOptions $ \options ->
+    RocksDB.withRocksDB options gopts.rootDir $ \db ->
+    RocksDB.checkpoint db $ gopts.rootDir ++ "_cp"
+
+doRun' :: GlobalOpts -> RunOpts -> IO ()
+doRun' gopts opts =
+    rocksDbWithOptions $ \options ->
+    RocksDB.withRocksDB options gopts.rootDir $ \db ->
+    rocksDbWithWriteOptions $ \wopts ->
+    RocksDB.withReadOptions $ \ropts ->
+        void $ forFoldM_ initGen [ 0 .. opts.batchCount - 1 ] $ \b g -> do
+            let lookups :: [Word64]
+                inserts :: [Word64]
+                (!nextG, lookups, inserts) = generateBatch gopts.initialSize opts.batchSize g b
+
+            -- lookups
+            let ks = makeKey <$> lookups
+
+            vs' <-
+                -- multi get or not.
+                if True
+                then RocksDB.multiGet db ropts ks
+                else forM ks (RocksDB.get db ropts)
+
+            vs <- evaluate (force vs')
+
+            -- check that we get values we expect
+            when opts.check $ do
```

Comment:
:+1: I was wondering if we were "testing" the rocksDB operations in any way, seems that you do